### PR TITLE
Auto reconnect to holochain

### PR DESCRIPTION
**Pull request by timotree3**
### Changes
- Move all the code from `websocket_wrappers` into `src/` and make them TypeScript
- Delete `ConnectionMonitor` and replace it with `reconnecting_websockets`
- Bump holochain-conductor-api dependency to a version that allows pluggable websocket backends (see https://github.com/holochain/holochain-conductor-api/pull/51)
- Add tests for the reconnection behavior
- Add type annotations to some envoy fields because I enjoy TypeScript :)
- Delete RPC Client code as it was unused (we use chaperone for unit tests instead) and depended on `ConnectionMonitor`

### CODE REVIEW

**Comment by timotree3** [comment]
Tests pass locally. Now only need a new release of `holochain-conductor-api` and to bump the dependency and CI should pass

File: src/index.ts [hunk 1]
```diff
@@ -143,37 +145,21 @@ class Envoy {
...
-        await client.opened(CONDUCTOR_TIMEOUT)
-          .catch(err => {
-            log.fatal("Conductor client '%s' failed to connect: %s", client.connectionMonitor.name, String(err));
-          });
-
-        log.debug("Conductor client '%s' is 'CONNECTED': readyState = %s", client.connectionMonitor.name, client.connectionMonitor.socket.readyState);
+      clients.map(async (pair) => {
+        const [name, client] = pair;
+        await client.opened();
+        log.debug("Conductor client '%s' is 'CONNECTED': readyState = %s", name, client.client.socket.readyState);
```
Comment (robbiecarlton) [inline src/index.ts:158]: 👍

File: src/websocket-wrappers/holochain.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+import Websocket from 'ws';
+
+import {WsClient as HolochainWsClient} from '@holochain/conductor-api/lib/websocket/client'
+
+import ReconnectingWebSocket from 'reconnecting-websocket';
+
+class HcAdminWebSocket extends AdminWebsocket {
+  constructor(url, ...args) {
+    super(new HolochainWsClient(new ReconnectingWebSocket(url, [], {
```
Comment (robbiecarlton) [inline src/websocket-wrappers/holochain.ts:19]: nice

**Review by robbiecarlton — APPROVED** [review]
This looks good. I'm not able to run the tests, so can you ask someone else to run them? Otherwise, good to go.

**Review by zo-el — APPROVED** [review]
Looks good. Great work with the Typing too :rocket: 

Also, we can close Issues https://github.com/Holo-Host/holo-envoy/issues/45

**Pull request merged**